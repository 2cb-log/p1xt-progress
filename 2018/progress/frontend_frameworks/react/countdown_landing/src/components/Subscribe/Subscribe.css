.Subscribe {
    text-align: left;
    width: 100vw;
    position: relative;
    height: 300px;
    background-color: #eee;
    background-attachment: fixed;
    background-repeat: no-repeat;
    background-position: 0 400px;
    background-size: cover;
    background-image: url('https://placekitten.com/1000/600?image=2');
}

.Subscribe__content {
    margin-top: 100px;
    position: relative;
    z-index: 5;
    color: #ffffff;
}

.Subscribe__content input {
    font-size: 16px;
    letter-spacing: 1px;
    font-weight: 300;
    color: #3b3b3b;
    width: 60%;
    height: 60px;
    padding: 0 20px;
    background-color: #fff;
    border: 1px solid #797979;
    border-right: none;
    border-radius: 4px;
    border-top-right-radius: 0px;
    border-bottom-right-radius: 0px;
}

.Subscribe__content button {
    font-size: 12px;
    letter-spacing: 1px;
    text-transform: uppercase;
    width: 30%;
    height: 60px;
    color: #ffffff;
    background: #c75e2d;
    border: 1px solid #797979;
    border-left: none;
    border-radius: 4px;
    border-top-left-radius: 0px;
    border-bottom-left-radius: 0px;
}

.Subscribe__content button:hover {
    background: #cf5821;
}