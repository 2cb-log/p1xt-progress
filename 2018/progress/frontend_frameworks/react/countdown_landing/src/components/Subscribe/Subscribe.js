import React from 'react';
import './Subscribe.css';

function Subscribe(props) {
  return (
    <div className="Subscribe">
      <div className="Subscribe__content container">
        <h1>Subscribe Now</h1>
        <input type="text" placeholder="Not collecting emails, demo only" />
        <button>Notify Me</button>
      </div>
    </div>
  );
}

export default Subscribe;
