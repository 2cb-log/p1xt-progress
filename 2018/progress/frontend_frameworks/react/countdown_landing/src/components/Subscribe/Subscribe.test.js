import React from 'react';
import ReactDOM from 'react-dom';
import Subscribe from './Subscribe';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Subscribe />, div);
  ReactDOM.unmountComponentAtNode(div);
});
it('renders correctly', () => {
  const tree = renderer
    .create(<Subscribe />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
