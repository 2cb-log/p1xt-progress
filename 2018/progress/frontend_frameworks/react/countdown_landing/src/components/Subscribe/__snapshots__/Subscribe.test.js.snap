// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Subscribe /> renders correctly 1`] = `
<div
  className="Subscribe"
>
  Subscribe
</div>
`;
